src/+mpecopt/Solver.m

```diff
@@ -934,12 +949,166 @@ function create_mpec_functions(obj)
+                            %         end
+                            %     end
+                            %     lbw = mpec.w.lb;
+                            %     lbw(ind_map_H) = 0;
+                            %     lbw(ind_map_G) = 0;
+                            %     mpec.w.lb = lbw;
+                            % end
+
+                            % TODO(@anton)
+                            %g_comp_expr = psi_fun(G_curr, H_curr, sigma);
+                            %[lb, ub, g_comp_expr] = generate_mpcc_relaxation_bounds(g_comp_expr, opts.relaxation_strategy);
+                            %mpec.g.(name)(curr{:}) = {g_comp_expr,lb,ub};
+                        end
+                    end
+                end
+                mpec.finalize_assignments();
+                new_w_indices = mpec.w.sort_by_index();
+                [~,new_map_w] = sort(new_w_indices);
+                new_g_indices = mpec.g.sort_by_index();
+                [~,new_map_g] = sort(new_g_indices);
+                ind_nonscalar_x1 = ind_nonscalar_x1;
+                ind_nonscalar_x2 = ind_nonscalar_x2;
+                dims.ind_x = new_map_w(1:n_primal_non_lifted);
+                dims.ind_g = new_map_g(1:n_g_non_lifted);
+                dims.map_w = new_map_w;
+                dims.map_g = new_map_g;
+
                 f = mpec.f;
+                x = mpec.w.sym;
                 g = mpec.g.sym;
+                p = mpec.p.sym;
                 G = mpec.G.sym;
                 H = mpec.H.sym;
-                p = mpec.p.sym;               
+
+                x1 = G;
+                x2 = H;
+                n_comp = length(x2);
+                
+                if n_comp > 0
+                    % TODO(@anton) do we actually need this here or can we calculate these "analytically"
+                    ind_x1 = [];
+                    ind_x2 = [];
+                    ind_x1_fun = Function('ind_1',{x},{x.jacobian(x1)});
+                    [ind_x1,~] = find(sparse(ind_x1_fun(zeros(size(x))))==1);
+                    ind_x2_fun = Function('ind_2',{x},{x.jacobian(x2)});
+                    [ind_x2,~] = find(sparse(ind_x2_fun(zeros(size(x))))==1);
+                    opts.nlp_is_mpec = 1; % TODO: check is this still used? (its purpose: if not an mpec, just make single nlp call without mpec machinery);
+                else
+                    opts.nlp_is_mpec = 0;
+                    ind_x1 = [];
+                    ind_x2 = [];
+                end
+                
+                n_primal = length(x);
+                n_primal_x0 = n_primal - 2*n_comp; % primal variables excluding the complementarity variables;
+                ind_x0 = [1:n_primal]';
+                if opts.nlp_is_mpec
+                    ind_x0([ind_x1,ind_x2]) = [];
+                end
+                x0 = x(ind_x0); % Variables not involved in complementarity constraints.
```

Comment:
In 1119 below (cant add comment to old core for some reason) - why the SX()?
Excludes MX models.

Reply:
Because otherwise passing just `[]` breaks things sometimes. There are a couple other places where we hard code use SX. I can go through and fix all of these in a future PR.

Reply:
i am ok with that. just open an issue so that we dont forget about it. thanks